app/controllers/add_products_to_carts_controller.rb

```diff
@@ -0,0 +1,65 @@
+class AddProductsToCartsController < ApplicationController
+	 
+
+	
+		
+
+	def update
+	
+		@product_detail = ProductDetail.find_by product_id: params[:id], size: params[:size], color: params[:color]
+	
+		id_product_detail = @product_detail.id
+		quantity_product_detail = params[:quantity]
+		@hash_content = current_cart.content
+	
+		if @hash_content.nil?
+			@hash_content ={"#{id_product_detail}": quantity_product_detail}
+			current_cart.update!(content: @hash_content)
+			
+		elsif @hash_content.has_key?("#{id_product_detail}")&& params[:cart_identy].nil?
+			
+			@old_quantity = current_cart.content["#{id_product_detail}"]
+			@hash_content[:"#{id_product_detail}"] = quantity_product_detail.to_i + @old_quantity.to_i
+			current_cart.update!(content: @hash_content)
+			flash.now[:success] = "them san pham thanh cong "
+		elsif !@hash_content.has_key?("#{id_product_detail}").nil?
```

Comment:
cái này có nghĩa là `@hash_content.has_key?("#{id_product_detail}")`
đúng k nhir?
vậy thì n là TH bao 2 trường hợp của dòng 19 và 31 hả anh?

Reply:
khong em, elsif !@hash_content.has_key?("#{id_product_detail}").nil? dung cho add to cart tai trang show product
con elsif @hash_content.has_key?("#{id_product_detail}")&& params[:cart_identy].nil? va  @hash_content.has_key?("#{id_product_detail}")&& params[:cart_identy].present? sử dụng cho trường hop thêm, giảm sản phẩm trong giỏ cart, lúc chưa di chuyển chuột ra khỏi nút +-